services/src/main/java/org/fao/geonet/api/mapservers/GeoFile.java

```diff
@@ -174,6 +176,29 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
                                 layers.add(base);
                             }
                         }
+                        if (fileIsGpkg(fileName)) {
+                            _format = "gpkg";
+                            String base = getBase(fileName);
+
+                            if (onlyOneFileAllowed) {
+                                if (layers.size() > 1)
+                                    throw new IllegalArgumentException(
+                                        "Only one geopackage per zip is allowed. "
+                                            + layers.size()
+                                            + " geopackage found.");
+
+                                if (base.equals(getBase(fileName))) {
```

Comment:
This check seems not required, from the previous assigment:

```
String base = getBase(fileName);
```

, it will be always `true`, no?

Reply:
that's just a copypaste from the section above handling shapefile, i think the logic was to ensure that `something.zip` only contained files named `something.ext`. So, to have a 'useful' check, i think it should be `base.equals(getBase(this.file.toString()))`. No ?

Reply:
I think so, please check also to update the shapefile code, thanks.

Reply:
trying `if (base.equals(getBase(this.file.toString())))`

i get this:
```
[ERROR] services/src/main/java/org/fao/geonet/api/mapservers/GeoFile.java:[167,61] cannot find symbol
[ERROR]   symbol: variable file
[ERROR] services/src/main/java/org/fao/geonet/api/mapservers/GeoFile.java:[190,61] cannot find symbol
[ERROR]   symbol: variable file
```

that's.. puzzling. why wouldnt the `simpleFileVisitor` be able to access `this.file` ? or `this` isnt the `GeoFile` object ? and i need to rename the local `file` object which shadows `GeoFile.file` ?

Reply:
That will not work, `this` in the `SimpleFileVisitor` refers to that class, not to `GeoFile`   you need to do something like this:

```
  public Collection<String> getVectorLayers(final boolean onlyOneFileAllowed) throws IOException {
        final LinkedList<String> layers = new LinkedList<String>();
        if (zipFile != null) {
            Path geoFile = this.file;
            
            ....

               if (base.equals(getBase(geoFile.toString()))) {
```

Reply:
yes, that's what i've figured out more or less. While trying to properly test this, i've also found out that the message/string sent with the `IllegalArgumentException` thrown wasn't used/displayed anywhere to the user, eg the `PUT /geonetwork/srv/api/mapservers/100/records/833fb482-1fdf-4ff1-9f24-837f3baeed34?metadataAbstract=Ceci+est+un+r%C3%A9sum%C3%A9&metadataTitle=geopublication+dallage+ortho&resource=xx.zip` returns a 404 with 
```
{"message":"GeoPublisherException","code" :"not_found","description":"Failed to publish vector file '/srv/data/geonetwork/data/metadata_data/00200-00299/242/public/xx.zip' Error is null"
```

it looks like some error reporting isnt properly passed over the codepaths. Will fix the test, wont be able to dig more into the underlying issue.

Reply:
after testing, the right check seems to be `if (base.equals(getBase(file.getFileName().toString())))` because `geoFile.file` is the full path.

using the api to check if a layer exists in the remote geoserver (with a "valid" zip) returns a proper json message with the 404:
```
{"message":"GeoPublisherException","code":"not_found","description":" Layer 'CAPI' not found in mapserver. Try to publish it? Status code is 404."}
```
using a non-valid zip (eg one which triggers the `IllegalArgumentException`) only yields:
```
{"message":"GeoPublisherException","code":"not_found","description":"Failed to publish vector file '/srv/data/geonetwork/data/metadata_data/00200-00299/242/public/xx.zip'. Error is null"}
```

Reply:
i've managed to fix the latter, which now properly returns the message propagated by the `IllegalArgumentException`:
```
{"message":"GeoPublisherException","code":"not_found","description":"Failed to publish vector file '/srv/data/geonetwork/data/metadata_data/00200-00299/242/public/xx.zip'. Error is Geopackage name (CAPI) is not equal to ZIP file name (xx.zip)."}

Reply:
09c50d3 & b97b2ef produce something that "feels" better.